src/coprocessor/dag/rpn_expr/impl_cast.rs

```diff
@@ -40,7 +43,8 @@ pub fn get_cast_fn(from_field_type: &FieldType, to_field_type: &FieldType) -> Re
         (EvalType::Duration, EvalType::Real) => cast_duration_as_real_fn_meta(),
         (EvalType::Json, EvalType::Real) => cast_json_as_real_fn_meta(),
         _ => return Err(box_err!("Unsupported cast from {} to {}", from, to)),
-    })
+    };
+    Ok((func, vec![ScalarArg::Int(0)]))
```

Comment:
Why add a new parameter here and it's always `[0]`?

Reply:
This cast function is inserted by Coprocessor automatically, the `inUnion` flag always false in this situation. Ideally, the cast function should be inserted by TiDB and pushed down with all implicit arguments.

Reply:
I was confused at first too. Could you add a comment here?

Reply:
OK